### PR TITLE
[extractor/tvp] fix live via vod

### LINKED ISSUE #8824: Can't download videos from vod.tvp.pl, e.g.: https://vod.tvp.pl/live,1/tvp-1,399697
### DO NOT REMOVE OR SKIP THE ISSUE TEMPLATE

- [X] I understand that I will be **blocked** if I *intentionally* remove or skip any mandatory\* field

### Checklist

- [X] I'm reporting that yt-dlp is broken on a **supported** site
- [X] I've verified that I have **updated yt-dlp to nightly or master** ([update instructions](https://github.com/yt-dlp/yt-dlp#update-channels))
- [X] I've checked that all provided URLs are playable in a browser with the same IP and same login details
- [X] I've checked that all URLs and arguments with special characters are [properly quoted or escaped](https://github.com/yt-dlp/yt-dlp/wiki/FAQ#video-url-contains-an-ampersand--and-im-getting-some-strange-output-1-2839-or-v-is-not-recognized-as-an-internal-or-external-command)
- [X] I've searched [known issues](https://github.com/yt-dlp/yt-dlp/issues/3766) and the [bugtracker](https://github.com/yt-dlp/yt-dlp/issues?q=) for similar issues **including closed ones**. DO NOT post duplicates
- [X] I've read the [guidelines for opening an issue](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#opening-an-issue)
- [ ] I've read about [sharing account credentials](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#are-you-willing-to-share-account-details-if-needed) and I'm willing to share it if required

### Region

Poland

### Provide a description that is worded well enough to be understood

Can't download videos from vod.tvp.pl, e.g.: https://vod.tvp.pl/live,1/tvp-1,399697
In the browser they play correctly.
Test on two different operating systems, nightly and stable version yt-dlp

### Provide verbose output that clearly demonstrates the problem

- [X] Run **your** yt-dlp command with **-vU** flag added (`yt-dlp -vU <your command line>`)
- [ ] If using API, add `'verbose': True` to `YoutubeDL` params instead
- [X] Copy the WHOLE output (starting with `[debug] Command-line config`) and insert it below

### Complete Verbose Output

```shell
[debug] Command-line config: ['--cookies-from-browser', 'firefox', 'https://vod.tvp.pl/live,1/tvp-2,399698', '-vU']
Extracting cookies from firefox
[debug] Extracting cookies from: "/home/manjaro/.mozilla/firefox/lmx686bz.default-release/cookies.sqlite"
Extracted 24 cookies from firefox
[debug] Encodings: locale UTF-8, fs utf-8, pref UTF-8, out utf-8, error utf-8, screen utf-8
[debug] yt-dlp version stable@2023.11.16 from yt-dlp/yt-dlp [24f827875]
[debug] Python 3.11.6 (CPython x86_64 64bit) - Linux-6.6.7-1-MANJARO-x86_64-with-glibc2.38 (OpenSSL 3.1.4 24 Oct 2023, glibc 2.38)
[debug] exe versions: rtmpdump 2.4
[debug] Optional libraries: certifi-2023.11.17, requests-2.31.0, secretstorage-3.3.3, sqlite3-3.44.2, urllib3-1.26.18
[debug] Proxy map: {}
[debug] Request Handlers: urllib, requests
[debug] Loaded 1901 extractors
[debug] Fetching release info: https://api.github.com/repos/yt-dlp/yt-dlp/releases/latest
Latest version: stable@2023.11.16 from yt-dlp/yt-dlp
yt-dlp is up to date (stable@2023.11.16 from yt-dlp/yt-dlp)
[tvp:vod] Extracting URL: https://vod.tvp.pl/live,1/tvp-2,399698
[tvp:vod] 399698: Downloading JSON metadata
ERROR: [tvp:vod] 399698: Woronicza said: ITEM_NOT_AVAILABLE (HTTP 404); please report this issue on  https://github.com/yt-dlp/yt-dlp/issues?q= , filling out the appropriate issue template. Confirm you are on the latest version using  yt-dlp -U
  File "/usr/lib/python3.11/site-packages/yt_dlp/extractor/common.py", line 715, in extract
    ie_result = self._real_extract(url)
                ^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/site-packages/yt_dlp/extractor/tvp.py", line 568, in _real_extract
    info_dict = self._parse_video(self._call_api(f'vods/{video_id}', video_id), with_url=False)
                                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/site-packages/yt_dlp/extractor/tvp.py", line 493, in _call_api
    raise ExtractorError(f'Woronicza said: {document.get("code")} (HTTP {urlh.status})')
```

### PR DESCRIPTION
**Pull request by selfisekai**
**IMPORTANT**: PRs without the template will be CLOSED

### Description of your *pull request* and other information

Fixes #8824

<details open><summary>Template</summary> 

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [contributing guidelines](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions) including [yt-dlp coding conventions](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#yt-dlp-coding-conventions)
- [x] [Searched](https://github.com/yt-dlp/yt-dlp/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [x] Checked the code with [flake8](https://pypi.python.org/pypi/flake8) and [ran relevant tests](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions)

### In order to be accepted and merged into yt-dlp each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check all of the following options that apply:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [x] Fix or improvement to an extractor (Make sure to add/update tests)
- [ ] New extractor ([Piracy websites will not be accepted](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#is-the-website-primarily-used-for-piracy))
- [ ] Core bug fix/improvement
- [ ] New feature (It is strongly [recommended to open an issue first](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#adding-new-feature-or-making-overarching-changes))

</details>

### CODE REVIEW

File: yt_dlp/extractor/tvp.py [hunk 1]
```diff
@@ -503,6 +506,14 @@ def _parse_video(self, video, with_url=True):
...
         })
         info_dict['description'] = clean_html(dict_get(video, ('lead', 'description')))
+        if type_ in ('PROGRAMME', 'LIVE'):
+            now = datetime.datetime.now().astimezone(datetime.timezone.utc).timestamp()
+            if parse_iso8601(video.get('since')) > now:
+                info_dict['live_status'] = 'is_upcoming'
+            elif video.get('till') is None or parse_iso8601(video.get('till')) >= now:
+                info_dict['live_status'] = 'is_live'
+            else:
+                info_dict['live_status'] = 'was_live'
```
Comment (seproDev) [inline yt_dlp/extractor/tvp.py:516]: I'm not familiar with the site, but by the looks of things, type `PROGRAMME` are the EPG entries. If we raise on date/don't even match date, is it even required to handle them here? Maybe I missed some other URL format that is also of the `PROGRAMME` type.

Could you also share an example of an upcoming/expired livestream?

**Review by seproDev — CHANGES_REQUESTED** [review]

File: yt_dlp/extractor/tvp.py [hunk 2]
```diff
@@ -560,12 +560,23 @@ class TVPVODVideoIE(TVPVODBaseIE):
             'thumbnail': 're:https?://.+',
         },
         'params': {'skip_download': 'm3u8'},
+    }, {
+        'url': 'https://vod.tvp.pl/live,1/tvp-world,399731',
+        'info_dict': {
+            'id': '399731',
+            'ext': 'mp4',
+            'title': r're:TVP WORLD(?: \d{4}-\d{2}-\d{2} \d{2}:\d{2})?',
```
Comment (seproDev) [inline yt_dlp/extractor/tvp.py:568]: Why is the date optional?
```suggestion
            'title': r're:TVP WORLD \d{4}-\d{2}-\d{2} \d{2}:\d{2}',
```

Comment (selfisekai) [reply yt_dlp/extractor/tvp.py:568]: i randomly got just "TVP WORLD" without a date during testing

Comment (bashonly) [reply yt_dlp/extractor/tvp.py:568]: Hmm, very weird. The test should fail in that case, though. We should be fixing the code instead of rigging the test

**Review by seproDev — APPROVED** [review]

**Review by bashonly — APPROVED** [review]

**Pull request merged**